flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/KeyedPartitionWindowedStreamITCase.java

```diff
@@ -97,6 +101,42 @@ public void mapPartition(
                 createExpectedString(3));
     }
 
+    @Test
+    void testRichMapPartitionFunctionHasOpen() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        DataStreamSource<Tuple2<String, String>> source = env.fromData(createSource());
+        source.keyBy(
+                        new KeySelector<Tuple2<String, String>, String>() {
+                            @Override
+                            public String getKey(Tuple2<String, String> value) throws Exception {
+                                return value.f0;
+                            }
+                        })
+                .fullWindowPartition()
+                .mapPartition(
+                        new RichMapPartitionFunction<Tuple2<String, String>, String>() {
+
+                            private boolean isOpen = false;
+
+                            @Override
+                            public void open(OpenContext openContext) throws Exception {
+                                super.open(openContext);
+                                isOpen = true;
+                            }
+
+                            @Override
+                            public void mapPartition(
+                                    Iterable<Tuple2<String, String>> values, Collector<String> out)
+                                    throws Exception {
+                                Assertions.assertTrue(
```

Comment:
We shall use `AssertJ` instead of `JunitAssertion`

`assertThat(isOpen).itTrue()`